src/sentry/static/sentry/app/views/organizationDiscover/result/pagination.jsx

```diff
@@ -11,33 +11,60 @@ export default class Pagination extends React.Component {
     getPreviousPage: PropTypes.func.isRequired,
     previous: PropTypes.object,
     next: PropTypes.object,
+    pageLimit: PropTypes.number.isRequired,
   };
 
+  getPageNumber() {
+    const {next, dataLength, pageLimit} = this.props;
+    const endRange = next.cursor.split(':')[1];
+
+    if (dataLength) {
+      const from = endRange - pageLimit + 1;
+      const to = dataLength < endRange ? from + dataLength : endRange;
+
+      return (
+        <NumberResultsShown>
+          Results {from} - {to}
+        </NumberResultsShown>
+      );
+    } else {
+      return <NumberResultsShown>0 Results</NumberResultsShown>;
+    }
+  }
 
   render() {
     const {getPreviousPage, getNextPage, previous, next} = this.props;
 
     return (
-      <PaginationButtons className="btn-group">
-        <Button
-          className="btn"
-          disabled={previous && !previous.results}
-          size="xsmall"
-          icon="icon-chevron-left"
-          onClick={getPreviousPage}
-        />
-        <Button
-          className="btn"
-          disabled={next && !next.results}
-          size="xsmall"
-          icon="icon-chevron-right"
-          onClick={getNextPage}
-        />
-      </PaginationButtons>
+      <React.Fragment>
+        <PaginationButtons className="btn-group">
+          <Button
+            className="btn"
+            disabled={previous && !previous.results}
+            size="xsmall"
+            icon="icon-chevron-left"
+            onClick={getPreviousPage}
+          />
+          <Button
+            className="btn"
+            disabled={next && !next.results}
+            size="xsmall"
+            icon="icon-chevron-right"
+            onClick={getNextPage}
+          />
+        </PaginationButtons>
+        {next && this.getPageNumber()}
```

Comment:
nit: I think we generally use a `render-` prefix for method names if they return a react component.